### PR TITLE
Add Cropping2D and Cropping3D

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?
This is to continue the work from https://github.com/intel-analytics/BigDL/pull/1946

## How was this patch tested?

new unit tests added

### CODE REVIEW

**Comment by hhbyyh** [comment]
cc @qiuxin2012

**Comment by hkvision** [comment]
Can this be merged?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping2D.scala [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    require(lenHCropped > 0, s"heightCrop: ${heightCrop.mkString(", ")} is too large. Height" +
+      s" dimension length: ${input.size(hdim)}")
+    require(lenWCropped > 0, s"widthCrop: ${widthCrop.mkString(", ")} is too large. Width" +
+      s" dimension length: ${input.size(wdim)}")
+
+    this.output = input
+      .narrow(hdim, hStart, lenHCropped)
+      .narrow(wdim, wStart, lenWCropped)
+      .contiguous()
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping2D.scala]: Please use resize & copy, contiguous will use new memory every time.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping2D.scala [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    val wStart = widthCrop(0) + 1
+    val lenWCropped = input.size(wdim) - widthCrop(0) - widthCrop(1)
+    (hdim, wdim, hStart, lenHCropped, wStart, lenWCropped)
+  }
+
+  override def clearState(): this.type = {
+    super.clearState()
+    this
+  }
+
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping2D.scala]: add equals and hashcode

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping3D.scala [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    require(dim2Cropped > 0, s"dim2Crop: ${dim2Crop.mkString(", ")} is too large. dim2" +
+      s" dimension length: ${input.size(dim2)}")
+    require(dim3Cropped > 0, s"dim3Crop: ${dim3Crop.mkString(", ")} is too large. dim3" +
+      s" dimension length: ${input.size(dim3)}")
+
+    this.output = input
+      .narrow(dim1, dim1Start, dim1Cropped)
+      .narrow(dim2, dim2Start, dim2Cropped)
+      .narrow(dim3, dim3Start, dim3Cropped)
+      .contiguous()
```
Comment (qiuxin2012) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cropping3D.scala]: like Cropping 2D

**Comment by hhbyyh** [comment]
updated

**Comment by hhbyyh** [comment]
cc @qiuxin2012 any other comments?

**Comment by qiuxin2012** [comment]
Jenkins passed?

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/job/BigDL-PR-V/562/ running. I'll merge it after validation passed.

**Comment by hkvision** [comment]
@hhbyyh style check failed la..

**Comment by hhbyyh** [comment]
got it.

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/job/BigDL-PR-V/567/

**Pull request merged**